### PR TITLE
Update ExemplarFilter and ExemplarReservoir for spec

### PR DESCRIPTION
**Pull request by ThomsonTan**
Fixes # (issue)

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [x] `CHANGELOG.md` updated for non-trivial changes
* [x] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

**Comment by marcalff** [comment]
Needs to be re evaluated after:

* https://github.com/open-telemetry/opentelemetry-specification/pull/3820

File: CHANGELOG.md [hunk 1]
```diff
@@ -65,6 +67,13 @@ Important changes:
     * Numbering of log levels like OTEL_INTERNAL_LOG_LEVEL_ERROR
       has changed, which requires to rebuild, as the SDK ABI differs.
 
+Notes on experimental features:
+
+* [#2327](https://github.com/open-telemetry/opentelemetry-cpp/issues/2327)
```
Comment (marcalff) [inline CHANGELOG.md:72]:
```suggestion
* [#2372](https://github.com/open-telemetry/opentelemetry-cpp/issues/2372)
```

**Review by marcalff — APPROVED** [review]
LGTM.

I fixed some typo in the PR number in CHANGELOG

File: CHANGELOG.md [hunk 2]
```diff
@@ -78,6 +78,8 @@ Increment the:
   [#2631](https://github.com/open-telemetry/opentelemetry-cpp/pull/2631)
 * [SDK] DefaultLogHandler to print errors to std::cerr, add LogLevel::None
   [#2622](https://github.com/open-telemetry/opentelemetry-cpp/pull/2622)
+* [SDK] Update ExemplarFilter and ExemplarReservoir for spec
```
Comment (marcalff) [inline CHANGELOG.md:81]: This CHANGELOG entry should go in the unreleased section.

File: CHANGELOG.md [hunk 3]
```diff
@@ -124,6 +126,13 @@ Important changes:
     * Numbering of log levels like OTEL_INTERNAL_LOG_LEVEL_ERROR
       has changed, which requires to rebuild, as the SDK ABI differs.
 
+Notes on experimental features:
+
+* [#2372](https://github.com/open-telemetry/opentelemetry-cpp/issues/2372)
+  introduced `MeterProvider::SetExemplar()` which accepts en
+  `ExemplarFilterType` enumeration with `kAlwaysOff`, `kAlwaysOn` and
+  `kTraceBased`.
+
```
Comment (marcalff) [inline CHANGELOG.md:135]: Likewise, move to unreleased section.

File: CHANGELOG.md (hunk 2 above)
Comment (ThomsonTan) [reply CHANGELOG.md:81]: Thanks for catching.

File: CHANGELOG.md (hunk 3 above)
Comment (ThomsonTan) [reply CHANGELOG.md]: Move both to unreleased section.

**Comment by ThomsonTan** [comment]
Talked with Lalit. He is fine to merge this PR and address potential feedback in future PRs.

I am going to merge it for now.

**Pull request merged**